crates/client/src/fuzzer/fuzzer_generator.rs

```diff
@@ -52,9 +53,25 @@ pub fn generate_source_code(idl: &Idl) -> String {
                             .parameters
                             .iter()
                             .map(|(name, ty)| {
-                                let name = format_ident!("{name}");
-                                let ty: syn::Type = parse_str(ty).unwrap();
-                                let parameter: syn::FnArg = parse_quote!(#name: #ty);
+                                let name_ident = format_ident!("{name}");
+                                // TODO What about custom Enums and Structs on Instr Input ?
+                                let ty = parse_str(ty).unwrap();
+                                let ty: syn::Type = match &ty {
```

Comment:
Why so complicated? You can compare the `ty` variable directly as it is a String before it is shadowed by the `let ty = parse_str(ty).unwrap();` line. So you can do only `if ty == "Pubkey"` directly. 

Also I think the accounts_from_instr_input HashMap is not necessary. If it is one of the accounts, that will be required by another instruction, than it will be automatically generated anyway. And if it is not an account used by any instruction, than there is no need to add it.

Reply:
Thanks for the comments. Removed the unnecessary HashMap.
Regarding the Pubkey parse, simplified with "ends_with". Long option "anchor_lang::pubkey::Pubkey" is stored inside ty as "anchor_lang :: pubkey :: Pubkey" so for that case I check for ":: Pubkey". I think it won't hurt to check also for "::Pubkey", and for the short path I check for "Pubkey"